app/src/app/utils/GeometryWorker/geometryWorker.ts

```diff
@@ -0,0 +1,110 @@
+import {expose} from 'comlink';
+import {area} from '@turf/area';
+import dissolve from '@turf/dissolve';
+import centerOfMass from '@turf/center-of-mass';
+import {GeometryWorkerClass} from './geometryWorker.types';
+import bboxClip from '@turf/bbox-clip';
+import pointOnFeature from '@turf/point-on-feature';
+import pointsWithinPolygon from '@turf/points-within-polygon';
+import {MapGeoJSONFeature} from 'maplibre-gl';
+
+const GeometryWorker: GeometryWorkerClass = {
+  geometries: {},
+  getGeos() {
+    return {
+      type: 'FeatureCollection',
+      features: Object.values(this.geometries),
+    };
+  },
+  updateProps(entries) {
+    entries.forEach(([id, zone]) => {
+      if (this.geometries[id]?.properties) {
+        this.geometries[id].properties['zone'] = zone;
+      }
+    });
+  },
+  loadGeometry(featuresOrStringified, idProp) {
+    const features: MapGeoJSONFeature[] =
+      typeof featuresOrStringified === 'string'
+        ? JSON.parse(featuresOrStringified)
+        : featuresOrStringified;
+    const firstEntry = Object.values(this.geometries)[0];
+    if (features.length && firstEntry) {
+      if (features[0].sourceLayer !== firstEntry.sourceLayer) {
+        this.geometries = {};
+      }
+    }
+    features.forEach(f => {
+      const id = f.properties?.[idProp];
+      if (id && !this.geometries[id]) {
+        this.geometries[id] = structuredClone(f);
+      }
+    });
+  },
+  dissolveGeometry(features) {
+    let dissolved: GeoJSON.FeatureCollection = dissolve(
+      {
+        type: 'FeatureCollection',
+        features: features.filter(
+          f => f.geometry.type === 'Polygon'
+        ) as GeoJSON.Feature<GeoJSON.Polygon>[],
+      },
+      {
+        propertyName: 'zone',
+      }
+    );
+    let largestDissolvedFeatures: Record<number, {feature: GeoJSON.Feature; area: number}> = {};
+
+    dissolved.features.forEach(feature => {
+      const zone = feature.properties?.zone;
+      if (!zone) return;
+      const featureArea = area(feature);
+      if (!largestDissolvedFeatures[zone] || featureArea > largestDissolvedFeatures[zone].area) {
```

Comment:
O: this makes sense for now given that we are not enforcing contiguity on zones, but could likely be refactored later when that rule is enforced

Reply:
Agreed! I added this note in a `TODO` along with a couple other future directions we could add